### PR TITLE
Fix #24, correct initialisation of elastic search

### LINKED ISSUE #24: v1.0.1 Elasticsearch __init__() throws error related to `user`
After merge of #17 I took master for a test drive 

It does not work for me when configured to use the ES (and Neo4j backend)

Looks like the named parameters signature here:

https://github.com/lyft/amundsensearchlibrary/blob/1a05eb2f063a0805d16261fcf4a6751023f502f9/search_service/proxy/elasticsearch.py#L33-L34

... should match `user` and `password` in the `__init__.py` file. 

I made the modification to Elasticsearch.py - and was able to make it work locally. Would like confirmation of if others see this as well.

Other than that I’m super excited about that #17 got merged for two reasons:

* Atlas with search (which I didn’t try yet) 
* That proxy is now configurable through env vars, with naming similar to that for metadata service, so config.py changes isn’t needed any more! 👍 @Whazor

### PR DESCRIPTION
**Pull request by Whazor**
### Summary of Changes

Fix for elastic search initialisation

### Tests

Added two tests, one to setup elastic search directly, one to set it up via the proxy method.

### Documentation

Not relevant

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`
- [x] I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

File: tests/unit/proxy/test_elasticsearch.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+
 from typing import Iterable
 
 from search_service import create_app
+from search_service.proxy import get_proxy_client
 from search_service.proxy.elasticsearch import ElasticsearchProxy
 from search_service.models.search_result import SearchResult
 from search_service.models.table import Table
 
-
```
Comment (jornh) [inline tests/unit/proxy/test_elasticsearch.py]: Seems Flake does want 2 lines here. Breaks in Travis

**Comment by feng-tao** [comment]
thanks @Whazor for the fix. There are still a few things missing:
1. Currently the index is set to None for the proxy init (https://github.com/lyft/amundsensearchlibrary/blob/master/search_service/proxy/__init__.py#L33) as opposed to https://github.com/lyft/amundsensearchlibrary/blob/master/search_service/proxy/elasticsearch.py#L284. We should instantiate the index if there is and pass to the proxy client constructor. Is atlas used any search index?
2. We should remove https://github.com/lyft/amundsensearchlibrary/blob/master/search_service/proxy/elasticsearch.py#L252-L290 given we don't use it any more.
3. I think we should rename every auth_user to user, and auth_pw to password in elasticsearchProxy.

**Comment by Whazor** [comment]
Atlas does not use search index, I moved it the config retrieval to ElasticSearch itself. The constructor argument is still there for the test cases. I also added the client_key method that was originally in ElasticSearch and the page size argument, which is also used in Atlas.

**Comment by feng-tao** [comment]
Thanks @Whazor . I think it LGTM. cc @jinhyukchang for another pair of eye.

**Comment by feng-tao** [comment]
I manually increase the version in the pr. Should be good.

**Comment by feng-tao** [comment]
thanks @Whazor for the fix :)

**Pull request merged**